btc-lsp/src/BtcLsp/Rpc/ScriptHashRpc.hs

```diff
@@ -0,0 +1,139 @@
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+
+module BtcLsp.Rpc.ScriptHashRpc
+  ( getBalance,
+    version,
+    Balance (..),
+    Address (..),
+    ScriptHash (..),
+  )
+where
+
+import BtcLsp.Import
+import BtcLsp.Rpc.Client as Client
+import BtcLsp.Rpc.RpcRequest as Req
+import BtcLsp.Rpc.RpcResponse as Resp
+import Data.Aeson (decode, encode, object, withObject, (.:), (.=))
+import qualified Data.ByteString.Lazy as BS
+import Data.Digest.Pure.SHA
+import Network.HTTP.Client as HTTPClient hiding (Proxy)
+import qualified Text.Hex as TH
+
+newtype Address = Address Text
+  deriving (Generic)
+
+newtype ScriptHash = ScriptHash Text
+  deriving (Generic)
+
+data Balance = Balance
+  { confirmed :: MSat,
+    unconfirmed :: MSat
+  }
+  deriving (Generic, Show)
+
+getBalanceFromSat :: MSat -> MSat -> Balance
+getBalanceFromSat c u = Balance (c * 1000) (u * 1000)
+
+instance FromJSON Balance where
+  parseJSON = withObject "Balance" $ \v ->
+    getBalanceFromSat
+      <$> v .: "confirmed"
+      <*> v .: "unconfirmed"
+
+instance ToJSON Address
+
+instance ToJSON ScriptHash
+
+callRpc :: (Env m, ToJSON req, FromJSON resp) => Method -> req -> m (Either RpcError resp)
+callRpc m r = do
+  let request =
+        RpcRequest
+          { Req.id = 0,
```

Comment:
What this id mean?

Reply:
_An identifier established by the Client that MUST contain a String, Number, or NULL value if included. If it is not included it is assumed to be a notification. The value SHOULD normally not be Null [[1]](https://www.jsonrpc.org/specification#id1) and Numbers SHOULD NOT contain fractional parts [[2]](https://www.jsonrpc.org/specification#id2)_

in electrs rpc example it's always 0 and not used for anything, but required)